core/activity/src/api.rs

```diff
@@ -22,32 +22,67 @@ mod common {
     use actix_web::{web, HttpResponse, Responder};
     use futures::prelude::*;
 
-    use ya_client_model::market::Role;
+    use ya_client_model::market::{Agreement, Role};
     use ya_core_model::{activity, NodeId};
     use ya_persistence::executor::DbExecutor;
     use ya_service_api_web::middleware::Identity;
     use ya_service_bus::{timeout::IntoTimeoutFuture, RpcEndpoint};
 
     use crate::common::*;
+    use crate::error::Error;
     use crate::tracker::TrackingEvent;
     use crate::TrackerRef;
     use actix_web::http::header;
+    use actix_web::web::Json;
+    use ya_client_model::payment::params::FilterParams;
 
     pub fn extend_web_scope(scope: actix_web::Scope) -> actix_web::Scope {
         scope
-            // .service(get_activities_web)
             .service(get_events)
+            .service(get_activities_web)
             .service(get_activity_agreement_web)
             .service(get_activity_state_web)
             .service(get_activity_usage_web)
     }
 
-    // TODO this endpoint needs authorization via Identity, otherwise is vulnerable for attacks.
-    // #[actix_web::get("/activity")]
-    // async fn get_activities_web(db: web::Data<DbExecutor>) -> impl Responder {
-    //     log::debug!("get_activities_web");
-    //     get_activities(&db).await.map(web::Json)
-    // }
+    #[actix_web::get("/activity")]
+    async fn get_activities_web(
+        db: web::Data<DbExecutor>,
+        query: web::Query<FilterParams>,
+        id: Identity,
+    ) -> impl Responder {
+        let mut result = vec![];
+        let activities = get_activities(&db, query.max_items).await?;
+
+        let is_valid = |agreement: Agreement| {
+            if authorize_caller(&id.identity.to_string().parse()?, agreement.provider_id()).is_ok()
+                && (query.after_timestamp.is_none()
+                    || query.after_timestamp.is_some()
+                        && agreement.timestamp >= query.after_timestamp.unwrap())
```

Comment:
As I commented below it is better to move timestamp filtering to database in this place